astropy/coordinates/tests/helper.py

```diff
@@ -11,7 +11,7 @@ def skycoord_equal(sc1, sc2):
         return False
     if sc1.shape != sc2.shape:
         return False  # Maybe raise ValueError corresponding to future numpy behavior
-    eq = np.ones(shape=sc1.shape, dtype=bool)
+    eq = True
     for comp in sc1.data.components:
-        eq &= getattr(sc1.data, comp) == getattr(sc2.data, comp)
-    return np.all(eq)
+        eq &= np.all(getattr(sc1.data, comp) == getattr(sc2.data, comp))
+    return eq
```

Comment:
The way it was set up, one started with an unmasked array of bool, but in-place `&=` on that with a masked array does not work (since the mask cannot be propagated to an unmasked-instance). So, this allows the code to work for masked components (properly ignoring elements that are masked).